sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Processor/PartitionLoadBalancer.cs

```diff
@@ -0,0 +1,443 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Messaging.EventHubs.Core;
+using Azure.Messaging.EventHubs.Errors;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    internal class PartitionLoadBalancer
+    {
+        /// <summary>The random number generator to use for a specific thread.</summary>
+        private static readonly ThreadLocal<Random> RandomNumberGenerator = new ThreadLocal<Random>(() => new Random(Interlocked.Increment(ref s_randomSeed)), false);
+
+        /// <summary>The seed to use for initializing random number generated for a given thread-specific instance.</summary>
+        private static int s_randomSeed = Environment.TickCount;
+
+        /// <summary>
+        ///   Interacts with the storage system with responsibility for creation of checkpoints and for ownership claim.
+        /// </summary>
+        ///
+        private readonly PartitionManager StorageManager;
+
+        /// <summary>
+        ///   The set of partition ownership this event processor owns.  Partition ids are used as keys.
+        /// </summary>
+        ///
+        private Dictionary<string, PartitionOwnership> InstanceOwnership { get; set; } = new Dictionary<string, PartitionOwnership>();
+
+        /// <summary>
+        ///   A partition distribution dictionary, mapping an owner's identifier to the amount of partitions it owns and its list of partitions.
+        /// </summary>
+        ///
+        private readonly Dictionary<string, List<PartitionOwnership>> ActiveOwnershipWithDistribution = new Dictionary<string, List<PartitionOwnership>>();
+
+        /// <summary>
+        ///   The fully qualified Event Hubs namespace that the processor is associated with.  This is likely
+        ///   to be similar to <c>{yournamespace}.servicebus.windows.net</c>.
+        /// </summary>
+        ///
+        public readonly string FullyQualifiedNamespace;
```

Comment:
I don't have a strong opinion, but you can feel free to leave these visible, make these `private` ,or encapsulate them into variables as you prefer.  The processor and other clients expose them for callers to be able to understand what is being processed (in case there are several being juggled or passed around into common methods).

Reply:
I needed them internal at least for tests, so I figured I could leave them visible for any other consumers as well.

Reply:
Fair enough.  Works for me.